### PR TITLE
Cast twitch's ID to float, since int was overflowing.

### PR DESCRIPTION
**Pull request by Nicarim**
Fixes situation where certain streams weren't possible to be highlighted.

### CODE REVIEW

**Comment by nanaya** [comment]
why not just string.

**Comment by Nicarim** [comment]
that was first that came to my mind, it doesn't matter that much at the end i guess.

**Comment by Tom94** [comment]
This is a bad idea. Floats have less precision than ints, even if they have higher maximum numbers. Read up on them on wikipedia if you want more details.

You really should use either string, or if those intgers were 32bit, use 64bit integers.

**Comment by nanaya** [comment]
(not related) and then javascript doesn't even have int :dancer:

File: app/Http/Controllers/CommunityController.php [hunk 1]
```diff
@@ -65,8 +65,9 @@ public function getLive(Request $request)
 
         $featuredStreamId = Cache::get('featuredStream');
         if ($featuredStreamId !== null) {
+            $featuredStreamId = floatval($featuredStreamId);
             foreach ($streams as $stream) {
-                if ($stream->_id !== $featuredStreamId) {
+                if (floatval($stream->_id) !== $featuredStreamId) {
```
Comment (nanaya) [inline app/Http/Controllers/CommunityController.php]: btw, if int overflows, what's in `$stream->_id` to begin with? If it's already int, I don't see how casting will help?

I think it's more like victim of strict comparison upgrade and `Cache` casts everything to string.

**Comment by Nicarim** [comment]
copy paste from slack:
sec.

[19:03] 
 ```      +"_id": 19511430960.0```<- twitch
```19511430960.0```
my casted float,

File: app/Http/Controllers/CommunityController.php (hunk 1 above)
Comment (Nicarim) [inline app/Http/Controllers/CommunityController.php]: Yeah, I think depending on cache driver, it casts to string. I also casted _id to float (but now I checked it, it is already float), just to be safe.

File: app/Http/Controllers/CommunityController.php [hunk 2]
```diff
@@ -84,7 +85,7 @@ public function postLive(Request $request)
         }
 
         if ($request->has('promote')) {
-            Cache::forever('featuredStream', intval($request->promote));
+            Cache::forever('featuredStream', floatval($request->promote));
```
Comment (nanaya) [inline app/Http/Controllers/CommunityController.php]: my guess says the cast will add '.0' to the stored string and thus will never pass the check. But maybe I'm wrong ¯\\\_(ツ)_/¯

Comment (peppy) [inline app/Http/Controllers/CommunityController.php]: i'd rather push to for 64-bit php for this project either way..

**Pull request merged**